### PR TITLE
added largeSync, lzString

### LINKED ISSUE #6: accurate byte count
https://github.com/dtuit/chrome-storage-largeSync

isolate this from HTML

### PR DESCRIPTION
**Pull request by stevekrouse**
These libraries are pretty small so I'm not too worried about adding them.

lz-string is popular and stable.

chrome-storage-largeSync is unpopular and the code confuses me but it seems to work well enough and fast enough and it's small enough. I've modified the code a bit to work better both in browser and chrome extension.

Closes #6 

I also did some little testing and it seems that the compression is good enough for us to do a really, really niave revision history (#14) simply by saving previous versions of a note as separate copies and then counting on it to shrink it down. (I'm not sure the speed implications of this yet.)

### CODE REVIEW

**Comment by stevekrouse** [comment]
**[This is an automated integration to preview this pull request's changes to the website.]**

https://rawcdn.githack.com/stevekrouse/blinknote/213d7195dbf332892ec2fad95ddc4206411d4696/index.html

File: index.js [hunk 1]
```diff
@@ -70,8 +64,8 @@ const title =  {
     title.state = state; 
     title.update();
   },
-  setCharacterLength: length => {
-    title.charLength = length
+  setCharacterLength: async length => {
+    title.charLength = await storage.getBytesInUse();
```
Comment (dankantor) [inline index.js]: Nice! Was going to say we should move to async/await from promises. If we do want to support this in all browsers, we may have to compile that however.

File: index.js [hunk 2]
```diff
@@ -40,28 +40,22 @@ const storage = {
...
       });
     }).then(storage.joinText),
   
   splitText: text => {
-    let splittedText = {};
-    storage.splitKeys().forEach(i => {
-      splittedText['text' + i] = text.substring(i * storage.QUOTA_BYTES_PER_ITEM, (i + 1) * storage.QUOTA_BYTES_PER_ITEM);
-    });
-    return splittedText;
+    return {text};
```
Comment (dankantor) [inline index.js]: `splitText` is not the best name for this method anymore. Maybe something like `textToObject`?

File: index.js [hunk 3]
```diff
@@ -89,14 +83,20 @@ const init = () => {
     chrome.storage = {
       sync: {
         get: (keys, callback) => {
-          callback(localStorage)
+          if (keys === null ){ keys = Object.keys(localStorage) }
+          let ret = {}
+          keys.forEach(key => ret[key] = JSON.parse(localStorage[key]))
+          callback(ret)
```
Comment (dankantor) [inline index.js]: We haven't discussed style yet. On line 93 you have a semicolon but here you don't.

File: index.js (hunk 2 above)
Comment (stevekrouse) [reply index.js]: yep!

File: index.js (hunk 1 above)
Comment (stevekrouse) [reply index.js]: It's a chrome extension ;)

File: index.js (hunk 3 above)
Comment (stevekrouse) [reply index.js]: How about prettier #20 as a pre-commit hook? I don't care about the setting so feel free to add your own config file

File: index.js (hunk 1 above)
Comment (dankantor) [reply index.js]: I wasn't sure if you wanted to move towards other browsers with `localStorage`. Or was that just for the githack?

Comment (stevekrouse) [reply index.js]: Ah yeah, that's just for githack previews

**Pull request merged**